includes/abstracts/abstract-wc-legacy-order.php

```diff
@@ -0,0 +1,621 @@
+
+		if ( 'completed_date' === $key ) {
+			return $this->get_date_completed();
+		} elseif ( 'paid_date' === $key ) {
+			return $this->get_date_paid();
+		} elseif ( 'modified_date' === $key ) {
+			return $this->get_date_modified();
+		} elseif ( 'order_date' === $key ) {
+			return $this->get_date_created();
+		} elseif ( 'id' === $key ) {
+			return $this->get_id();
+		} elseif ( 'post' === $key ) {
+			return get_post( $this->get_id() );
+		} elseif ( 'status' === $key || 'post_status' === $key ) {
+			return $this->get_status();
+		} elseif ( 'customer_message' === $key || 'customer_note' === $key ) {
+			return $this->get_customer_note();
+		} elseif ( in_array( $key, array( 'user_id', 'customer_user' ) ) ) {
+			return $this->get_customer_id();
+		} elseif ( 'tax_display_cart' === $key ) {
+			return get_option( 'woocommerce_tax_display_cart' );
+		} elseif ( 'display_totals_ex_tax' === $key ) {
+			return 'excl' === get_option( 'woocommerce_tax_display_cart' );
+		} elseif ( 'display_cart_ex_tax' === $key ) {
+			return 'excl' === get_option( 'woocommerce_tax_display_cart' );
+		} elseif ( 'cart_discount' === $key ) {
+			return $this->get_discount();
+		} elseif ( 'cart_discount_tax' === $key ) {
+			return $this->get_discount_tax();
+		} elseif ( 'order_tax' === $key ) {
+			return $this->get_cart_tax();
+		} elseif ( 'order_shipping_tax' === $key ) {
+			return $this->get_shipping_tax();
+		} elseif ( 'order_shipping' === $key ) {
+			return $this->get_shipping_total();
+		} elseif ( 'order_total' === $key ) {
+			return $this->get_total();
+		} elseif ( 'order_type' === $key ) {
+			return $this->get_type();
+		} elseif ( 'order_currency' === $key ) {
+			return $this->get_currency();
+		} elseif ( 'order_version' === $key ) {
+			return $this->get_version();
+	 	} elseif ( is_callable( array( $this, "get_{$key}" ) ) ) {
+			return $this->{"get_{$key}"}();
+		} else {
+			return get_post_meta( $this->get_id(), '_' . $key, true );
+		}
+	}
+
+	/**
+	 * has_meta function for order items.
+	 *
+	 * @param string $order_item_id
+	 * @return array of meta data.
+	 */
+	public function has_meta( $order_item_id ) {
+		global $wpdb;
+
+		_deprecated_function( 'has_meta', '2.7', 'WC_Order_item::get_meta_data' );
+
+		return $wpdb->get_results( $wpdb->prepare( "SELECT meta_key, meta_value, meta_id, order_item_id
+			FROM {$wpdb->prefix}woocommerce_order_itemmeta WHERE order_item_id = %d
+			ORDER BY meta_id", absint( $order_item_id ) ), ARRAY_A );
+	}
+
+	/**
+	 * Display meta data belonging to an item.
+	 * @param  array $item
+	 */
+	public function display_item_meta( $item ) {
+		_deprecated_function( 'get_item_meta', '2.7', 'wc_display_item_meta' );
+		$product   = $item->get_product();
+		$item_meta = new WC_Order_Item_Meta( $item, $product );
+		$item_meta->display();
+	}
+
+	/**
+	 * Display download links for an order item.
+	 * @param  array $item
+	 */
+	public function display_item_downloads( $item ) {
```

Comment:
Should this throw a deprecated notice and suggest using `wc_display_item_downloads()` instead?